include/seqan3/alignment/configuration/align_config_band.hpp

```diff
@@ -14,51 +14,104 @@
- *
- * \tparam band_t The type of the band.
  *
  * \details
  *
- * Configures the banded alignment algorithm. Currently only seqan3::static_band is allowed as argument.
- * If no band is configured for the alignment algorithm the full alignment matrix will be computed.
- * Before executing the algorithm the band is tested for valid settings, e.g. that the upper bound is not smaller than
- * the lower bound, or the band is not shifted out of the alignment matrix. If an invalid setting is detected, a
- * seqan3::invalid_alignment_configuration exception will be thrown.
+ * Configures the banded alignment algorithm. Currently only a fixed size band is allowed.
+ * The band is given in form of a seqan3::align_cfg::lower_diagonal and a seqan3::align_cfg::upper_diagonal.
+ * A diagonal represents the cells in the alignment matrix that are not crossed by the alignment either downwards by
+ * the lower diagonal or rightwards by the upper diagonal. Thus any computed alignment will be inside the area defined
+ * by the lower and the upper diagonal.
+ *
+ * If this configuration is default constructed or not set during the algorithm configuration the the full alignment
+ * matrix will be computed.
+ *
+ * During the construction of this configuration element or before the execution of the alignment algorithm the
+ * band configuration is validated. If the user provided an invalid band, e.g. the upper diagonal is smaller than
+ * the lower bound, or the alignment matrix would be ill configured such that the requested alignment method cannot
+ * be computed (because the global alignment requires the first cell and the last cell of the matrix to be reachable),
+ * then a seqan3::invalid_alignment_configuration will be thrown.
  *
  * ### Example
  *
  * \include test/snippet/alignment/configuration/align_cfg_band_example.cpp
  */
-template <typename band_t>
-//!\cond
-    requires std::same_as<band_t, static_band>
-//!\endcond
-struct band : public pipeable_config_element<band<band_t>, band_t>
+class band_fixed_size : public pipeable_config_element<band_fixed_size, seqan3::detail::empty_type>
 {
+private:
+    //!\brief The base class type.
+    using base_t = pipeable_config_element<band_fixed_size, seqan3::detail::empty_type>;
+public:
+
+    //!\brief The selected lower diagonal. Defaults to 0.
+    seqan3::align_cfg::lower_diagonal lower_diagonal{std::numeric_limits<int32_t>::lowest()};
+    //!\brief The selected upper diagonal. Defaults to 0.
+    seqan3::align_cfg::upper_diagonal upper_diagonal{std::numeric_limits<int32_t>::max()};
+
+    constexpr band_fixed_size() = default; //!< Defaulted.
+    constexpr band_fixed_size(band_fixed_size const &) = default; //!< Defaulted.
+    constexpr band_fixed_size(band_fixed_size &&) = default; //!< Defaulted.
+    constexpr band_fixed_size & operator=(band_fixed_size const &) = default; //!< Defaulted.
+    constexpr band_fixed_size & operator=(band_fixed_size &&) = default; //!< Defaulted.
```

Comment:
```suggestion
    band_fixed_size() = default; //!< Defaulted.
    band_fixed_size(band_fixed_size const &) = default; //!< Defaulted.
    band_fixed_size(band_fixed_size &&) = default; //!< Defaulted.
    band_fixed_size & operator=(band_fixed_size const &) = default; //!< Defaulted.
    band_fixed_size & operator=(band_fixed_size &&) = default; //!< Defaulted.
```

Reply:
nop. they are declared constexpr on purpose: 
So something like this is possible: `inline constexpr configuration my_config = config1 | config2`. etc.

Reply:
Ah ok sorry. Is it tested?

Reply:
Btw. All the selectors will have a std::variant which should be constexpr since c++17 but is not in GCC7. So a lot of configurations will not be possible at constexpr

Reply:
I am aware of that, but then we need to ugily make it just a const variable in our code as long as we support gcc-7. That doesn't mean that the user who should use a modern compiler cannot do it the right way.